subsys/net/lib/http/http_client.c

```diff
@@ -717,7 +717,15 @@ int http_client_req(int sock, struct http_request *req,
 	total_recv = http_wait_data(sock, req, timeout);
 	if (total_recv < 0) {
 		NET_DBG("Wait data failure (%d)", total_recv);
+		ret = total_recv;
+		goto out;
 	} else {
+		if (total_recv == 0) {
```

Comment:
Could this just be an `else if () {` instead, given there's goto inside? IMO that'd be slightly more readable.

Reply:
Yes, good point, I will change this.